pkg/poll/poll.go

```diff
@@ -411,14 +411,27 @@ func ForBuild(obj *unstructured.Unstructured) error {
 		return errors.Wrap(err, "Could not get BuildList")
 	}
 
-	for _, build := range builds.Items {
-		callback := makeStatusCallback(&build, "Complete", "status", "phase")
-		if err := ForResourceFullAvailability(&build, callback); err != nil {
+	var build *unstructured.Unstructured
+	for _, b := range builds.Items {
+		slice, _, err := unstructured.NestedSlice(b.Object, "metadata", "ownerReferences")
+		if err != nil {
 			return err
 		}
+		for _, element := range slice {
+			if name, ok := element.(map[string]interface{})["name"]; ok && name == obj.GetName() {
```

Comment:
If the key does not exist in the map, the value will be the empty string. I'd say `ok` is redundant here.
```suggestion
			if element.(map[string]interface{})["name"]== obj.GetName() {
```